polaris/hub/polarisfs.py

```diff
@@ -212,3 +239,30 @@ def pipe_file(
             timeout=timeout,
         )
         response.raise_for_status()
+
+        if self.use_listings_cache:
+            new_listing = [{"name": path, "type": "file", "size": len(content)}]
+            current_listings = self.dircache.get(self._parent(path), None)
+            if current_listings is not None:
+                new_listing.extend(current_listings)
+                self.dircache[self._parent(path)] = new_listing
+            else:
+                self.dircache[self._parent(path)] = new_listing
```

Comment:
Good thinking to add this here!

Can be simplified a bit if we assume `self.use_listings_cache` to always be True. 

```suggestion
        new_listing = [{"name": path, "type": "file", "size": len(content)}]
        parent_path = self._parent(path)
        listings = self.dircache.get(parent_path, [])
        listings.append(new_listing)
        self.dircache[parent_path] = listings
```